x-pack/plugins/security_solution/public/network/pages/index.tsx

```diff
@@ -34,6 +33,12 @@ const NetworkContainerComponent = () => {
     [capabilitiesFetched, userHasMlUserPermissions]
   );
 
+  const getPathWithFlowType = useCallback(
+    (detailName: string) =>
+      `${NETWORK_PATH}/ip/${detailName}/${FlowTarget.source}/${NetworkRouteType.flows}`,
+    []
+  );
```

Comment:
Current URLs don't have the last `tabName` part. So if users go to `ip/10.8.104.155/destination`we should redirect to `ip/10.8.104.155/destination/flow` , the current implementation redirects to `ip/10.8.104.155/source/flow`. We have to fix it.
I think that adding the last parameter as optional in the Route path definition: `${NETWORK_PATH}/ip/:detailName/:flowTarget?`, and using it in this function if defined should be enough

Reply:
I tried to implement this cleanly but the Route doesnt cleanly have access to the params of its children. I came across this:
https://github.com/remix-run/react-router/issues/7803, specifically this comment: https://github.com/remix-run/react-router/issues/7803#issuecomment-871586384

In short, useParams, or more applicable in this case- the match object doesnt have access to the needed parameter at this level, to pass it to the redirect.

V6 of react router has a fix for this but I think v5 is left wanting. 

That said I think it should be possible with useMatchPath, or a custom function but I cant seem to get the former to work, and the latter seems messy 🤔 . Can we chat about this in the am.

This is what I was trying though with no luck along with variations:

```
  const location = useLocation();  // location  -> /network/ip/:detailName/:flowTarget/:tabName
  const match = matchPath(location.pathname, {   
    path: `/${NETWORK_PATH}/ip/:detailName/:flowTarget/:tabName`,
    exact: false,
    strict: false,
  });
 
  console.log(match); // null
```

Reply:
oh! So Ive thought that using match object here was dependent upon having access to the parameter from within the parent component. I had a complete misunderstanding 🤦. Thanks for showing this.